src/payment/strategies/mollie/mollie-script-loader.ts

```diff
@@ -0,0 +1,27 @@
+import { ScriptLoader } from '@bigcommerce/script-loader';
+
+import { PaymentMethodClientUnavailableError } from '../../errors';
+
+import { MollieClient, MollieHostWindow } from './mollie';
+
+export default class MollieScriptLoader {
+    constructor(
+        private _scriptLoader: ScriptLoader,
+        private _window: MollieHostWindow = window
+    ) {}
+
+    load(merchantId: string, locale: string, testmode: boolean): Promise<MollieClient> {
+        return this._scriptLoader
+            .loadScript('https://js.mollie.com/v1/mollie.js')
+            .then(() => {
+                if (!this._window.Mollie) {
```

Comment:
probably you can use a custom typeguard here instead
```
if (!isMollieWindow(this._window)) {
  // throw
}

```

Reply:
if i add this typeguard for some reason i get this warning

_**Cannot invoke an object which is possibly 'undefined'.**_